js/contact.js

```diff
@@ -281,4 +284,31 @@ function appendContactLink(contact) {
 	contactLanding.appendChild(contactLink);
 
 	return contactLink;
+}
+
+/**
+ * Sorts all contacts by first name
+ */
+function sortContacts() {
+	let contacts = document.getElementById("contactLanding");
```

Comment:
Luckily, you don't need to know any sorting algorithms to sort things in js. All you need is a comparator, or in other words, something to tell js what is *greater* and what is *less* and it'll do the rest for you:

```javascript
// Sorting by object's name
myCollection.sort((obj1, obj2) => (obj1.name > obj2.name) ? 1 : -1);
```

When it comes to content of sorting, you should sort the contacts map, and then repopulate the DOM based on those values. This saves a lot of DOM class lookups and makes things more concise overall.

Reply:
This is awesome! I had no idea that maps were even sortable. I had always thought they were meant to be an unordered datastructure. Thank you!

Reply:
The maps themselves aren’t sortable but you can sort their contents into an array

Reply:
Got it :+1:

Reply:
This seems to work for me:

```javascript
const alphaComparator = (contact1, contact2) => (contact1.FirstName > contact2.FirstName) ? 1 : -1;

 // It's a map so it nests its array, hence the [0] subscript.
const sorted = new Array([...contacts.values()].sort(alphaComparator))[0];
```

Reply:
I have something very similar to this in essence. We are now using the sort function as our default for displaying contacts, as we want them to always be in sorted order. The displaying must also happen AFTER the sorting, which is why I did this.

Reply:
Unfortunately, this style seems to break the auto selecting feature I built in a while ago, fixing now

Reply:
Maybe decouple sorting from the populating method and pass the sorted array into instead